mirrord/layer/src/lib.rs

```diff
@@ -832,8 +831,9 @@ pub(crate) unsafe extern "C" fn close_nocancel_detour(fd: c_int) -> c_int {
 #[cfg(target_os = "linux")]
 #[hook_guard_fn]
 pub(crate) unsafe extern "C" fn uv_fs_close(a: usize, b: usize, fd: c_int, c: usize) -> c_int {
+    let res = FN_UV_FS_CLOSE(a, b, fd, c);
     close_layer_fd(fd);
-    FN_UV_FS_CLOSE(a, b, fd, c)
+    res
```

Comment:
The documentation of `close_layer_fd` says it should be called after the original close.

Reply:
just remembered - the reason I did this this way is because calling `FN_UV_FS_CLOSE` causes it to not return to us ever afaik (due to event loop fun)

Reply:
Ok, so I'm reverting this change and adding a comment. Is `close_layer_fd` fine with being called before in this case?

Reply:
yeah, it's overall okay to do so it's just being "tidy" (freeing resources in the reverse order of being acquired, though there's no real constraint)